src/test/java/org/folio/rest/unit/ItemDamagedStatusAPIUnitTest.java

```diff
@@ -0,0 +1,333 @@
+import org.folio.rest.persist.PostgresClient;
+import org.folio.rest.persist.cql.CQLWrapper;
+import org.folio.rest.persist.interfaces.Results;
+import org.folio.rest.support.PostgresClientFactory;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.stubbing.Answer;
+
+import com.github.mauricio.async.db.postgresql.exceptions.GenericDatabaseException;
+import com.github.mauricio.async.db.postgresql.messages.backend.ErrorMessage;
+
+import io.vertx.core.AsyncResult;
+import io.vertx.core.Context;
+import io.vertx.core.Future;
+import io.vertx.core.Handler;
+import io.vertx.ext.unit.Async;
+import io.vertx.ext.unit.TestContext;
+import io.vertx.ext.unit.junit.RunTestOnContext;
+import io.vertx.ext.unit.junit.VertxUnitRunner;
+
+@RunWith(VertxUnitRunner.class)
+public class ItemDamagedStatusAPIUnitTest {
+  private static final String DEFAULT_QUERY = "";
+  private static final int DEFAULT_OFFSET = 0;
+  private static final int DEFAULT_LIMIT = 10;
+  private static final String DEFAULT_LANGUAGE = "en";
+  private static final HashMap<String, String> DEFAULT_HEADERS = new HashMap<>();
+  private static final String INTERNAL_SERVER_ERROR_MSG = "Oops! Some error was happened on the server side.";
+  @Rule
+  public RunTestOnContext rule = new RunTestOnContext();
+  @Mock
+  private PostgresClientFactory pgClientFactory;
+  @Mock
+  private PostgresClient postgresClient;
+  @InjectMocks
+  private ItemDamagedStatusAPI itemDamagedStatusAPI = new ItemDamagedStatusAPI();
+
+
+  @Before
+  public void setUp(TestContext testContext) throws Exception {
+    MockitoAnnotations.initMocks(this);
+    when(pgClientFactory.getInstance(any(Context.class), any(Map.class))).thenReturn(postgresClient);
+  }
+
+  @Test
+  public void getItemDamagedStatusesShouldReturnBadRequest(TestContext testContext) {
+    RuntimeException exception = new RuntimeException(INTERNAL_SERVER_ERROR_MSG);
+    doAnswer(setExceptionForHandlerArgument(6, exception))
+      .when(postgresClient)
+      .get(
+        eq(REFERENCE_TABLE),
+        eq(ItemDamageStatus.class),
+        eq(new String[]{"*"}),
+        any(CQLWrapper.class),
+        eq(true),
+        eq(true),
+        any(Future.class)
+      );
+
+    Async async = testContext.async();
+    Future<Response> responseFuture = Future.future();
+    itemDamagedStatusAPI.getItemDamagedStatuses(
+      DEFAULT_QUERY,
+      DEFAULT_OFFSET,
+      DEFAULT_LIMIT,
+      DEFAULT_LANGUAGE,
+      DEFAULT_HEADERS,
+      responseFuture.completer(),
+      rule.vertx().getOrCreateContext()
+    );
+
+    responseFuture.setHandler(it -> {
+      testContext.assertEquals(400, it.result().getStatus());
+      async.complete();
+    });
+  }
+
+  @Test
+  public void getItemDamagedStatusesShouldReturnInternalServerError(TestContext testContext) {
```

Comment:
The test name does not describe why an internal server error should be returned. Under what circumstances should it return an internal server error, when a specific exception is thrown?

Reply:
Renamed